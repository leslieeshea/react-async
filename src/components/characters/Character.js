import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  const {
    name,
    status,
    species,
    image
  } = character;

  return (
    <section>
      <img src={image} />
      <p>{name}</p>
      <p>{status}</p>
      <p>{species}</p>
    </section>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Character;
