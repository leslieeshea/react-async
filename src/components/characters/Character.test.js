import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const character = {
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      image: '/image'
    };

    const wrapper = shallow(<Character character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
