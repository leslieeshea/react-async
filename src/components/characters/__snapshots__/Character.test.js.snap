// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<section>
  <img
    src="/image"
  />
  <p>
    Morty Smith
  </p>
  <p>
    Alive
  </p>
  <p>
    Human
  </p>
</section>
`;
